### PR TITLE
BIGTOP-3522. Drop Fedora 31 and add Fedora 33 support.

### PR DESCRIPTION
**Pull request by sekikn**
https://issues.apache.org/jira/browse/BIGTOP-3522

### CODE REVIEW

**Comment by sekikn** [comment]
I tested this PR as follows. For the same reason as BIGTOP-3492, the repository URL in config_fedora-33.yaml is invalid until 3.0.0 is released, so an incompatible but existent URL is passed to docker-provisioner via command line option here.

```
$ cd docker/bigtop-puppet
$ ./build.sh trunk-fedora-33
$ cd ../bigtop-slaves
$ ./build.sh trunk-fedora-33
$ cd ../..
$ ./gradlew bigtop-utils-pkg-ind zookeeper-pkg-ind repo-ind -POS=fedora-33

...

BUILD SUCCESSFUL in 12m 18s
3 actionable tasks: 3 executed
$ ./gradlew docker-provisioner \
>   -Pconfig=config_fedora-33.yaml \
>   -Penable_local_repo \
>   -Pnum_instances=1 \
>   -POS=fedora-33 \
>   -Prepository='http://repos.bigtop.apache.org/releases/1.5.0/fedora/31/$basearch' \
>   -Prun_smoke_tests \
>   -Psmoke_tests=zookeeper \
>   -Pstack=bigtop-utils,zookeeper

...

TestZookeeper > testZkServerStatus STANDARD_ERROR
    21/03/09 00:08:50 INFO lang.Object: Running zkServer.sh status
    21/03/09 00:08:50 INFO lang.Object: zkServer.sh status checks out.
Finished generating test XML results (0.009 secs) into: /bigtop-home/bigtop-tests/smoke-tests/zookeeper/build/test-results/test
Generating HTML test report...
Finished generating test html results (0.062 secs) into: /bigtop-home/bigtop-tests/smoke-tests/zookeeper/build/reports/tests/test
Now testing...
:bigtop-tests:smoke-tests:zookeeper:test (Thread[Daemon worker,5,main]) completed. Took 2.168 secs.

BUILD SUCCESSFUL in 2m 5s
39 actionable tasks: 7 executed, 32 up-to-date
Stopped 1 worker daemon(s).
+ rm -rf buildSrc/build/test-results/binary
+ rm -rf /bigtop-home/.gradle

BUILD SUCCESSFUL in 4m 3s
1 actionable task: 1 executed
```

File: provisioner/docker/config_fedora-33.yaml [hunk 1]
```diff
@@ -15,9 +15,9 @@
 
 docker:
         memory_limit: "4g"
-        image: "bigtop/puppet:trunk-fedora-31"
+        image: "bigtop/puppet:trunk-fedora-33"
```
Comment (iwasakims) [inline provisioner/docker/config_fedora-33.yaml:18]: I could not pull this image yet.

File: bigtop_toolchain/manifests/packages.pp [hunk 2]
```diff
@@ -34,7 +34,6 @@
         "fuse-devel",
         "cppunit-devel",
         "openssl-devel",
-        "python2-pip",
```
Comment (iwasakims) [inline bigtop_toolchain/manifests/packages.pp:37]: We do not python2-pip for CentOS 7 and CentOS 8?

File: provisioner/docker/config_fedora-33.yaml (hunk 1 above)
Comment (sekikn) [reply provisioner/docker/config_fedora-33.yaml:18]: After merging this, we can run the Jenkins job that builds and pushes a new image with this PR. Then it becomes available on DockerHub.

File: bigtop_toolchain/manifests/packages.pp (hunk 2 above)
Comment (sekikn) [reply bigtop_toolchain/manifests/packages.pp:37]: I think we can drop pip2 from CentOS 7 and 8 too, for the same reason as #728.

**Review by iwasakims — APPROVED** [review]
+1.

**Comment by sekikn** [comment]
@iwasakims Thank you for the review! Just merged.

**Pull request merged**